src/utility.ts

```diff
@@ -27,6 +28,56 @@ class Utility {
   private static strBreak =
     "************************************************************";
 
+  /**
+   * Checks to see if 'xdg-open' is on the path
+   */
+  isXdgOpenAvailable(): boolean {
+    if (this.isFileOnPath("xdg-open")) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * For each platform, create a way to open a file with the filetypes
+   * default application.
+   *
+   */
+  openWithDefaultApp(file: string) {
+    switch (os.platform()) {
+      case "win32": {
+        spawnSync(`start ${file}`);
+        break;
+      }
+      case "darwin": {
+        spawnSync(`open ${file}`);
+        break;
+      }
+      case "linux": {
+        if (this.isXdgOpenAvailable()) {
+          spawnSync("xdg-open");
+        } else {
+          const errorMsgs: string[] = [
+            "XDG-OPEN needs to be installed to launch embedded links.",
+          ];
+
+          outputChannel.appendError(Utility.strBreak);
+
+          for (const m of errorMsgs) {
+            outputChannel.appendError(m);
+          }
```

Comment:
don't need the string array, for loop, or join below for a single message

Reply:
Copy and paste, what a waste.